drawers/cAllegroDrawer.cpp

```diff
@@ -200,6 +202,18 @@ void cAllegroDrawer::drawRectangleFilled(BITMAP *dest, const cRectangle &pRectan
     rectfill(dest, pRectangle.getX(), pRectangle.getY(), pRectangle.getEndX(), pRectangle.getEndY(), color);
 }
 
+void cAllegroDrawer::drawRectangleTransparentFilled(BITMAP *dest, const cRectangle& rect, int color, int alpha) {
+  assert(alpha >= 0);
+  assert(alpha <= 255);
+
+  // Create a temporary bitmap that automatically destroys itself when going out-of-scope
+  std::unique_ptr<BITMAP, decltype(&destroy_bitmap)> rect_bmp{create_bitmap(rect.getWidth(), rect.getHeight()), destroy_bitmap};
```

Comment:
nice intermediate solution.

I was thinking if we could optimize this. Ie, I notice the rectangle (bitmap) we create is often the same dimension(s). So we might even be able to cache these created bitmaps for re-usage (to re-draw) and only destroy them once the `cAllegroDrawer` gets destroyed.

It saves create/deletion every frame. 

So basically we get a map I suppose, where the key could be `<width>x<height>` and the value the `BITMAP *`?

Reply:
Not a bad idea, actually.